### PR TITLE
Change design of lists in web UI

### PR DESCRIPTION
**Pull request by Gargron**
This should make creating and managing lists a lot more pleasant.

https://github.com/user-attachments/assets/83c1164a-8054-44f9-892d-ddcfb8635974

### CODE REVIEW

File: app/javascript/svg-icons/squiggly_arrow.svg
Comment (mjankowski) [inline app/javascript/svg-icons/squiggly_arrow.svg:1]: Excellent filename.

**Comment by ThisIsMissEm** [comment]
What's the "show replies to" thing? That kind of makes it sounds like the list may be public? Maybe "Show in the list replies to"

**Comment by ThisIsMissEm** [comment]
On the profile list modal, maybe it makes sense to have a footer of "Manage all your lists here" type thing? Since the list management UI is buried under the `[...]` next to the username above the composer.

**Comment by ThisIsMissEm** [comment]
Also, are we being consistent in the placement of the action buttons in modals? I think some are at the bottom, some are in the header like here? It may be good to pick one and stick with it.

**Comment by ThisIsMissEm** [comment]
And final question: what happens if I click a username whilst managing a list? Does that give me a profile overview modal or something, or does it wipe out the fact that I'm managing a list at the moment and jump me to their profile page?

**Comment by ClearlyClaire** [comment]
> What's the "show replies to" thing? That kind of makes it sounds like the list may be public? Maybe "Show in the list replies to"

It's the same thing as we currently have, which allows you to chose which replies appear in the list *for you*:
image

I'm not sure how this makes it sound like the list may be public?

File: app/javascript/mastodon/features/lists/index.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+          </Link>
+        }
+      />
+
+      <ScrollableList
+        scrollKey='lists'
+        emptyMessage={emptyMessage}
+        bindToDocument={!multiColumn}
+      >
+        {lists.map((list: List) => (
```
Comment (renchap) [inline app/javascript/mastodon/features/lists/index.tsx:132]: Why do you need to specify the type here?

File: app/javascript/styles/mastodon/components.scss [hunk 2]
```diff
@@ -11150,3 +11080,104 @@ noscript {
...
+
+    &:focus {
+      outline: 0;
+    }
+  }
+}
+
+.column-footer {
+  box-sizing: border-box;
+  position: fixed;
```
Comment (ClearlyClaire) [inline app/javascript/styles/mastodon/components.scss:11162]: This unfortunately has a few issues:
- it is completely broken in the advanced interface
- it hides the last item of the list when scrolling is needed

File: app/javascript/styles/mastodon/components.scss [hunk 3]
```diff
@@ -2760,7 +2760,8 @@ a.account__display-name {
       flex: 0 1 auto;
       display: flex;
       flex-direction: column;
-      contain: inline-size layout paint style;
+
+      // contain: inline-size layout paint style; // FIXME: breaks position: fixed
```
Comment (renchap) [inline app/javascript/styles/mastodon/components.scss:2764]: Needs to be fixed?

**Review by renchap — COMMENTED** [review]
First review purely looking at the code, I did not try to run it

**Review by ClearlyClaire — COMMENTED** [review]
I have not fully reviewed the code, but barring the CSS bugs I posted, the UX is a clear improvement.

I do have some (mostly technical) worry about the `Add @name to lists` modal though: checking/unchecking a list is immediately applied, which is not entirely obvious, and is also likely to be expensive, especially with #32930. It *might* make more sense to apply the changes in batch after a “Done” button is pressed or something.

**Comment by ThisIsMissEm** [comment]
> > What's the "show replies to" thing? That kind of makes it sounds like the list may be public? Maybe "Show in the list replies to"
> 
> It's the same thing as we currently have, which allows you to chose which replies appear in the list _for you_: image
> 
> I'm not sure how this makes it sound like the list may be public?

I think it could definitely do with a clearer label. I was certainly confused by this when looking at it (I don't use lists much personally)

**Comment by ClearlyClaire** [comment]
Maybe instead of “Show replies from list members to”, have “Include replies from list members to”?

**Comment by Gargron** [comment]
All addressed except for the changes to "Add {name} to lists" modal. It might make sense to do this but I'd rather flag it down to be done separately, add a rate limit or debouncing.

**Review by ClearlyClaire — DISMISSED** [review]
I have only skimmed over the code changes, but I have found no issue when trying it out, it appears functional, and is a clear design improvement.

I'll defer to @renchap for the actual code review, but looks good to me otherwise.

**Review by renchap — APPROVED** [review]

**Comment by mjankowski** [comment]
Something I noticed ... as the screen gets taller, the fun squiggly arrow gets further away from what its pointing at...

Screenshot 2024-11-24 at 09 48 11

Could see adjusting that to be dropped some standard distance from the top, instead of vertical centered?

**Pull request merged**